director/director_advertise_test.go

```diff
@@ -22,21 +22,108 @@ package director_test
 
 import (
 	"encoding/json"
+	"io"
 	"net/http"
 	"net/http/httptest"
 	"sync/atomic"
 	"testing"
+	"time"
 
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/pelicanplatform/pelican/fed_test_utils"
+	"github.com/pelicanplatform/pelican/param"
 	"github.com/pelicanplatform/pelican/server_structs"
 	"github.com/pelicanplatform/pelican/server_utils"
 )
 
+// Test for a director disappearing
+func TestDirectorShutdown(t *testing.T) {
+	server_utils.ResetTestState()
+
+	var listDirectorCount atomic.Int32
+	dirAd := &server_structs.DirectorAd{}
+	dirAd.Initialize("fake-director")
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		log.Debugln("Fake director received", req.Method, "for path", req.URL.Path)
+		if req.Method == "GET" && req.URL.Path == "/api/v1.0/director/directors" {
```

Comment:
I count three instances of nearly identical http test servers that mock the Director like this. At that point, is there any benefit in adding a helper function to our `test_utils` package?

Reply:
Agreed -- let's refactor.

I added this to the list of follow-up items: https://github.com/PelicanPlatform/pelican/issues/2119